site/docs/concepts/high-level-vs-low-level.md

```diff
@@ -13,6 +13,18 @@ The first statement, is described in a high-level abstraction layer. The higher
 
 The flexibility of IaSQL– which makes it really awesome– is that you can create both the high-level and the low-level modules for it. Both of the above statements are already possible using IaSQL modules. The first one using a module named `aws_ecs_simplified` and the latter using the `aws_security_group` module.
 
+Low-level modules provide building blocks that can be used by the high-level modules. In this case, `aws_security_group` defines these two entities:
+- [SecurityGroup](https://github.com/iasql/iasql-engine/blob/main/src/modules/0.0.22/aws_security_group/entity/index.ts#L19)
+- [SecurityGroupRule](https://github.com/iasql/iasql-engine/blob/main/src/modules/0.0.22/aws_security_group/entity/index.ts#L70)
+
+Which have their own tables in the IaSQL database (`security_group` and `security_group_rule`). These tables can be easily used for CRUD operations on those entities. For example, if you execute an `INSERT` statement in the `security_group` table followed by an `iasql_apply()` RPC, a new security group will be created in your AWS account. The logic for creation of a `SecurityGroup` entity in the cloud is also [handled by the `aws_security_group` module](https://github.com/iasql/iasql-engine/blob/main/src/modules/0.0.22/aws_security_group/index.ts#L219). So this module is providing a great building block for doing all the CRUD operations on `SecurityGroup` entities on the cloud.
+
+Now comes in the `aws_ecs_simplified` high-level module. It takes the advantage of the CRUD operations handled by the `aws_security_group` module. So when you insert a new record into the `ecs_simplified` table (to create an ECS app with all the needed resources), that `INSERT` command [triggers](https://github.com/iasql/iasql-engine/blob/main/src/modules/0.0.22/aws_ecs_simplified/sql/after_install.sql#L159-L165) a function that [inserts records for a new security group with two rules](https://github.com/iasql/iasql-engine/blob/main/src/modules/0.0.22/aws_ecs_simplified/sql/after_install.sql#L64-L73) into the tables managed by `aws_security_group`. So in the next `iasql_apply()`, the `aws_security_group` module will provision that security group and add those two rules to it.
```

Comment:
The only nit is that all of these links to source code should be changed to the `v0.0.22` tag instead of `main` once `v0.0.22` is released to prevent link-rot.